src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/NoOpRestoreUtilities.cs

```diff
@@ -158,6 +161,34 @@ internal static bool VerifyRestoreOutput(RestoreRequest request, CacheFile cache
                 return false;
             }
 
+            if (request.ForceUpdatePackageLastAccessTime)
+            {
+                foreach (var package in cacheFile.ExpectedPackageFilePaths)
+                {
+                    var packageFolder = new DirectoryInfo(Directory.GetParent(package).FullName).Parent.Parent.FullName;
+
+                    // we avoid fallback folders as they are readonly
+                    if (request.Project.RestoreMetadata.FallbackFolders.Any(fallbackFolder => PathUtility.GetStringComparerBasedOnOS().Equals(fallbackFolder, packageFolder)))
+                        continue;
+
+                    var packageId = new DirectoryInfo(Directory.GetParent(package).FullName).Parent.Name;
+                    var version = Directory.GetParent(package).Name;
+                    var resolver = new VersionFolderPathResolver(packageFolder);
+                    var metadataFile = resolver.GetNupkgMetadataPath(packageId, new NuGetVersion(version));
+
+                    try
+                    {
+                        File.SetLastAccessTimeUtc(metadataFile, DateTime.UtcNow);
+                    }
+                    catch(Exception ex)
+                    {
+                        await request.Log.LogAsync(RestoreLogMessage.CreateWarning(NuGetLogCode.NU1802,
```

Comment:
Use the sync method here, and the whole method back to sync. We don't need the additional allocations.